### PR TITLE
Refactor and add tests for git package

### PR DESCRIPTION
**Pull request by Warashi**
**What this PR does / why we need it**:

This PR does four things to improve git package.
1. adds test for repository copy method.
2. remove unused private method `getLatestRemoteHashForBranch` from git client
3. fix the variable used in test because it seems typo
4. test not private method but its exported wrapper method

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**: No

- **How are users affected by this change**: N/A
- **Is this breaking change**: N/A
- **How to migrate (if breaking change)**: N/A

### CODE REVIEW

**Review by t-kikuc — APPROVED** [review]
Thank you!!  What a clear commit log!

**Review by ffjlabo — APPROVED** [review]
Thank you!

File: pkg/git/repo_test.go [hunk 1]
```diff
@@ -206,7 +206,7 @@ func Test_setGCAutoDetach(t *testing.T) {
 	}
 
 	getGCAutoDetach := func(ctx context.Context, repo *repo) (bool, error) {
-		cmd := exec.CommandContext(ctx, r.gitPath, "config", "--get", "gc.autoDetach")
+		cmd := exec.CommandContext(ctx, repo.gitPath, "config", "--get", "gc.autoDetach")
```
Comment (khanhtc1202) [inline pkg/git/repo_test.go:209]: @Warashi Sorry for the late comment, look like the `r` variable in the below line (L210) has the same issue with this fixed one. It works because both refer to the outer `r` variable 👀 Would be great if you could open another PR to fix this.

Comment (Warashi) [reply pkg/git/repo_test.go:209]: I opened another PR to fix this with other improvements.
https://github.com/pipe-cd/pipecd/pull/4932
